app/uk/gov/hmrc/customs/notification/repo/LockRepo.scala

```diff
@@ -18,31 +18,38 @@ package uk.gov.hmrc.customs.notification.repo
 
 import java.util.UUID
 
-import org.joda.time.Duration
-import play.api.libs.json.{JsValue, Json}
+import javax.inject.{Inject, Singleton}
+import org.joda.time.{DateTime, DateTimeZone, Duration}
+import play.api.libs.json.Json
 import reactivemongo.api.DB
 import uk.gov.hmrc.customs.notification.domain.ClientSubscriptionId
+import uk.gov.hmrc.customs.notification.logging.NotificationLogger
 import uk.gov.hmrc.lock.LockFormats.{Lock, expiryTime}
 import uk.gov.hmrc.lock.{ExclusiveTimePeriodLock, LockFormats, LockRepository}
-import uk.gov.hmrc.mongo.CurrentTime
 import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
 case class LockOwnerId(id: String) extends AnyVal
 
-trait LockRepo extends CurrentTime{
+@Singleton
+class LockRepo @Inject()(mongoDbProvider: MongoDbProvider,
+                         notificationLogger: NotificationLogger) {
+
+  val repo = new LockRepository()(mongoDbProvider.mongo)
+
+  protected lazy val zone = DateTimeZone.UTC
+
+  private def withCurrentTime[A](f: (DateTime) => A) = f(DateTime.now.withZone(zone))
```

Comment:
Duplicated code, this code is already available in uk.gov.hmrc.mongo.CurrentTime